### ISSUE TITLE
Multisite functions not available pre-install

### ISSUE THREAD

**Issue by joehoyle** [humanmade/altis-core#53]
Currently the following will produce a fatal on install, _and_ requests to `wp-admin/install.php` if you put this in your `.config/load.php`:

```php
add_action( 'plugins_loaded', function () {
    var_dump( get_site()->public );
} );
```

This is because multisite is not turned on pre-install, and not loaded on `install.php` even if the site _is_ installed. This leads to bad fatals, as if someone has the site setup, the above will work without issues. We also need to fix the `wp-admin/install.php` requests, as that's currently is what Cloud uses for a healthcheck.

```
php_1            | [23-Aug-2019 20:42:32 UTC] PHP Fatal error:  Uncaught Error: Call to undefined function get_site() in /usr/src/app/.config/constants.php:24
php_1            | Stack trace:
php_1            | #0 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(286): private_sites_require_login('')
php_1            | #1 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters(NULL, Array)
php_1            | #2 /usr/src/app/wordpress/wp-includes/plugin.php(465): WP_Hook->do_action(Array)
php_1            | #3 /usr/src/app/wordpress/wp-settings.php(329): do_action('muplugins_loade...')
php_1            | #4 /usr/src/app/wp-config.php(67): require_once('/usr/src/app/wo...')
php_1            | #5 /usr/src/app/wordpress/wp-load.php(42): require_once('/usr/src/app/wp...')
php_1            | #6 /usr/src/app/wordpress/wp-admin/install.php(36): require_once('/usr/src/app/wo...')
php_1            | #7 {main}
php_1            |   thrown in /usr/src/app/.config/constants.php on line 24
php_1            | 172.19.0.13 -  23/Aug/2019:20:42:32 +0000 "GET /wordpress/wp-admin/install.php" 500
```

**Comment by joehoyle** [comment]
There might be a case for rolling our own `wp-settings.php` here, so we can get the multisite stuff always-on, pretty sure this is blocked by the `is_multisite()` check in that file at the moment.

**Comment by rmccue** [comment]
This seems very low priority here, is this actually causing any problems in production?

**Comment by joehoyle** [comment]
Yeah this has caused issues for a deploy already. I think the sooner we fix the better as it might require changing our load order of things

-- 

Joe Hoyle

CTO
Human Made

**Comment by rmccue** [comment]
@joehoyle Can you go into a bit more detail here about how we'd go about resolving this? Seems like this bug would occur in regular WordPress as well, I don't see what's specific to our setup here.

**Comment by joehoyle** [comment]
We're currently recommending `.config/load.php` is used for procedural configuration and per-site level configuration. As Altis is "always multisite", we should be providing that MS api in all contexts.

At the moment, a developer _can_ put: 

```php
add_action( 'plugins_loaded', function () {
    var_dump( get_site()->public );
} );
``` 

In `.config/load.php`, and it works, no issues, then they push to production and the deploy will fail because in "weird" contexts like `wp-admin/install.php` (which is our Cloud healthcheck), and installing the site a new, the MS functions will not be available. It's a broken support and experience for `.config/load.php` (and really any code that assumed multisite).

WordPress core is "multisite option" so you write accordingly, Altis is "MS always on" so should be relied upon.

As for a fix, I think it should be easy enough to modify the bootstrap to always include multisite functions even in the event of the networking not being installed, or `install.php` being specifically excluded from loading the MS api.

**Comment by roborourke** [comment]
Gonna take this out of the 3.0 milestone. The immediate issues are resolved for now and rolling a custom `wp-settings.php` will be a bunch of work and yet another thing to maintain.

**Comment by roborourke** [comment]
Now that we have `WP_INITIAL_INSTALL` I think we can close this. The usage of it is documented in one of the guides but at any rate the specific issue we raised here can be worked around using that constant.
